core/services/job/orm.go

```diff
@@ -132,13 +139,27 @@ func (o *orm) CreateJob(ctx context.Context, jobSpec *models.JobSpecV2, taskDAG
 	defer cancel()
 
 	return postgres.GormTransaction(ctx, o.db, func(tx *gorm.DB) error {
+		// TODO(connor): this needs to accept the tx to be transactional?
 		pipelineSpecID, err := o.pipelineORM.CreateSpec(ctx, taskDAG)
 		if err != nil {
 			return errors.Wrap(err, "failed to create pipeline spec")
 		}
 		jobSpec.PipelineSpecID = pipelineSpecID
 
 		err = tx.Create(jobSpec).Error
+		if err != nil {
+			if strings.Contains(err.Error(), `violates foreign key constraint "offchainreporting_oracle_specs_p2p_peer_id_fkey"`) {
```

Comment:
I believe the gorm error object actually contains a `ConstraintName` field or something similar which might be easier to use and more reliable

Reply:
done

Reply:
cool didn't know it returned pq.Error types